locallib.php

```diff
@@ -450,10 +449,25 @@ function moodleoverflow_print_forum_list($course, $cm, $movetopopup) {
     $forumarray = [[]];
     $currentforum = $DB->get_record('moodleoverflow_discussions', ['id' => $movetopopup], 'moodleoverflow');
     $currentdiscussion = $DB->get_record('moodleoverflow_discussions', ['id' => $movetopopup], 'name');
-    $forums = $DB->get_records('moodleoverflow', ['course' => $course->id]);
+
+    // If the currentforum is anonymous, only show forums that have a higher anonymous setting.
+    $anonymoussetting = $DB->get_field('moodleoverflow', 'anonymous', ['id' => $currentforum->moodleoverflow]);
+    if ($anonymoussetting == anonymous::QUESTION_ANONYMOUS || $anonymoussetting == anonymous::EVERYTHING_ANONYMOUS) {
```

Comment:
It should be possible to move a post from a forum where questions are anonymous to a forum where questions are anonymous correct? Currently only everything anonymous forums are displayed.

Reply:
Yes it should be this way, it's fixed now.